craft_providers/pro.py

```diff
@@ -19,23 +19,32 @@
 import json
 import logging
 from pathlib import Path
+from typing import Tuple
 
 import requests
 
-from craft_providers.errors import MachineTokenError
+from craft_providers.errors import GuestTokenError, MachineTokenError
 
 logger = logging.getLogger(__name__)
 
 
-def retrieve_pro_host_token() -> str:
-    """Get the machine token from the pro host."""
+def retrieve_pro_host_info() -> Tuple[str, str, str]:
```

Comment:
What do you think about a dataclass? Now that this is returning 3 pieces of data for the same object, it would be a cleaner interface.

```python
@dataclass
class ProHostInfo:
    machine_token: str
    machine_id: str
    contract_id: str
```